compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6453,10 +6454,14 @@ public void visit(BLangInvocation.BLangActionInvocation actionInvocation) {
             transactionDesugar.startTransactionCoordinatorOnce(env, actionInvocation.pos);
         }
 
-        // Add `@strand {thread: "any"}` annotation to an isolated start-action or
-        // isolated named worker declaration.
-        if (actionInvocation.async && Symbols.isFlagOn(actionInvocation.symbol.type.flags, Flags.ISOLATED)) {
-            addStrandAnnotationWithThreadAny(actionInvocation);
+        // Add `@strand {thread: "any"}` annotation to an isolated start-action in an isolated function.
+        if (!actionInvocation.functionPointerInvocation && actionInvocation.async &&
+                Symbols.isFlagOn(actionInvocation.symbol.type.flags, Flags.ISOLATED) &&
+                isInIsolatedFunction(env.enclInvokable)) {
```

Comment:
Here I think we don't need to check whether the start action is inside an isolated function. I think what is mentioned in the spec is that isolated functions can only contain isolated start actions.
Strands created by the isolated start actions can run on a separate thread even when it is not inside an isolated function. So I think we can omit the `&& isInIsolatedFunction(env.enclInvokable)` part.

Reply:
Btw previously even though we added `@strand {thread: "any"}` annotation to the `functionPointerInvocation start actions` also, it did not cause a problem because it got omitted when creating a FPcall. Either way if we have `!actionInvocation.functionPointerInvocation &&` part or not, I think it is fine, but need to do the most suitable thing. I think we can remove that check also.

Reply:
> Here I think we don't need to check whether the start action is inside an isolated function. I think what is mentioned in the spec is that isolated functions can only contain isolated start actions. Strands created by the isolated start actions can run on a separate thread even when it is not inside an isolated function. So I think we can omit the `&& isInIsolatedFunction(env.enclInvokable)` part.
 
Fixed this. 


> Btw previously even though we added `@strand {thread: "any"}` annotation to the `functionPointerInvocation start actions` also, it did not cause a problem because it got omitted when creating a FPcall. Either way if we have `!actionInvocation.functionPointerInvocation &&` part or not, I think it is fine. I think we can remove that check also.

Unnecessarily adding an annotation would cause an overhead, i think, because we do some desugaring related to the record init function introduced with the annotation in the closure desugar. Hence keeping this as it is.